sinkhole-core/src/server_elgamal.rs

```diff
@@ -0,0 +1,150 @@
+mod traits;
+
+use errors::errors::ServerError;
+use server_elgamal::traits::*;
+use std::cell::RefCell;
+use rand_core::OsRng;
+
+use curve25519_dalek::scalar::Scalar; 
+use curve25519_dalek::constants::RISTRETTO_BASEPOINT_TABLE;
+use elgamal_ristretto::private::SecretKey;
+use elgamal_ristretto::public::PublicKey;
+use elgamal_ristretto::ciphertext::Ciphertext;
+
+struct EGServer {
+    secretKey: SecretKey,
+    size: usize,
+    store: RefCell<Vec<Scalar>>,
+}
+
+impl EGServer {
```

Comment:
If we move this stuff to an Elgamal specific directory we also don't need the prefix.